provider/cmd/pulumi-gen-aws-native/main.go

```diff
@@ -30,10 +30,14 @@ import (
 	"github.com/pulumi/pulumi-aws-native/provider/pkg/refdb"
 	"github.com/pulumi/pulumi-aws-native/provider/pkg/schema"
 	pschema "github.com/pulumi/pulumi/pkg/v3/codegen/schema"
-	"github.com/pulumi/pulumi/sdk/v3/go/common/tools"
 	"github.com/pulumi/pulumi/sdk/v3/go/common/util/contract"
 )
 
+// ensureDir ensures that a target directory exists (like `mkdir -p`), returning a non-nil error if any problem occurs.
```

Comment:
Looks like it's a 1:1 copy of the removed pu/pu code. LGTM!
https://github.com/pulumi/pulumi/blob/1837a707182a1a9e2f9ed4898895e5bc562dbaa4/sdk/go/common/tools/paths.go#L26-L29